mapsims/noise.py

```diff
@@ -0,0 +1,194 @@
+        apply_kludge_correction : bool
+            If True, reduce the hitcount by a factor of 0.85 to account for not-uniformity in the scanning
+        scanning_strategy : str
+            Choose between the available scanning strategy hitmaps "classical" or "opportunistic" or
+            path to another hitmap
+        LA_number_LF : int
+            Number of Low Frequency tubes in LAT
+        LA_number_MF : int
+            Number of Medium Frequency tubes in LAT
+        LA_number_UHF : int
+            Number of Ultra High Frequency tubes in LAT
+        SA_years_LF : int
+            Number of years for the Low Frequency detectors to be deployed on the Small Aperture telescopes
+        SA_one_over_f_mode : {"pessimistic", "optimistic", "none"}
+            Correlated noise performance of the detectors on the Small Aperture telescopes
+        """
+
+        self.telescope = telescope
+        assert telescope in ["LA", "SA"]
+        self.band = int(band)
+        self.sensitivity_mode = sensitivity_modes[sensitivity_mode]
+        self.apply_beam_correction = apply_beam_correction
+        self.apply_kludge_correction = apply_kludge_correction
+        self.nside = nside
+        self.return_uK_CMB = return_uK_CMB
+        self.ell_max = 3 * nside
+        self.LA_number_LF = LA_number_LF
+        self.LA_number_MF = LA_number_MF
+        self.LA_number_UHF = LA_number_UHF
+        self.SA_years_LF = SA_years_LF
+        self.SA_one_over_f_mode = one_over_f_modes[SA_one_over_f_mode]
+        self.SA_remove_kluge = SA_remove_kluge
+
+        # Load hitmap and compute sky fraction
+
+        if os.path.exists(scanning_strategy):
+            hitmap_filename = scanning_strategy
+        else:
+            hitmap_filename = data.get_pkg_data_filename(
+                "data/total_hits_{}_{}.fits.gz".format(
+                    self.telescope, scanning_strategy
+                )
+            )
+        self.hitmap = hp.ud_grade(
+            hp.read_map(hitmap_filename, verbose=False), nside_out=self.nside
+        )
+        self.hitmap /= self.hitmap.max()
+        # Discard pixels with very few hits that cause border effects
+        self.hitmap[self.hitmap < 1e-3] = 0
+        self.sky_fraction = (self.hitmap != 0).sum() / len(self.hitmap)
+
+        if self.telescope == "SA":
+            self.ell, self.noise_ell_P, _ = so_noise.Simons_Observatory_V3_SA_noise(
+                self.sensitivity_mode,
+                self.SA_one_over_f_mode,
+                self.SA_years_LF,
+                self.sky_fraction,
+                self.ell_max,
+                delta_ell=1,
+                apply_beam_correction=self.apply_beam_correction,
+                apply_kludge_correction=self.apply_kludge_correction,
+            )
+            # For SA, so_noise simulates only Polarization,
+            # Assume that T is half
+            self.noise_ell_T = self.noise_ell_P / 2
```

Comment:
I doubt we'll care about T, but that's fine

Reply:
(just a reminder to remember T for anything LAT-related!)